tasks/basecalling/task_dorado_basecall.wdl

```diff
@@ -0,0 +1,53 @@
+version 1.0
+
+task basecall {
+  input {
+    File input_file                    # Single POD5 file for scatter processing
+    String dorado_model = "sup"         # Default model to 'sup', can be overridden with full model name see docs
+    String kit_name                    # Sequencing kit name
+    Int disk_size = 100
+    Int memory = 32
+    Int cpu = 8
+    String docker = "us-docker.pkg.dev/general-theiagen/staphb/dorado:0.8.0"
+  }
+
+  command <<< 
+    set -e
+
+    # Create output directory for SAM files
+    sam_output="output/sam/"
+    mkdir -p "$sam_output"
+
+    echo "### Starting basecalling for ~{input_file} ###"
+
+    base_name=$(basename "~{input_file}" .pod5)
+    sam_file="$sam_output/${base_name}.sam"
+
+    echo "Processing ~{input_file}, output: $sam_file"
+
+    # Run Dorado basecaller
+    dorado basecaller \
+      "~{dorado_model}" \
+      "~{input_file}" \
+      --kit-name ~{kit_name} \
+      --emit-sam \
+      --no-trim \
```

Comment:
Is there any reason why we are defaulting to `--no-trim`?

Might be good to set this as a string input in case we want to enable trimming in the future?

in the inputs:
`String? trim_mode = "none" # options are: "none", "all", "adapters", "primers"`

and in the command block:
`--trim ~{trim_mode} \`

Here's the relevant part of the `dorado basecaller --help` output:
```
Trimming arguments (detailed usage):
  --no-trim                   Skip trimming of barcodes, adapters, and primers. If option is not chosen, trimming of all three is enabled. 
  --trim                      Specify what to trim. Options are 'none', 'all', 'adapters', and 'primers'. Default behaviour is to trim all detected adapters, primers, or barcodes. Choose 'adapters' to just trim adapters. The 'primers' choice will trim adapters and primers, but not barcodes. The 'none' choice is equivelent to using --no-trim. Note that this only applies to DNA. RNA adapters are always trimmed. [nargs=0..1] [default: ""]
```

Reply:
It needs to be no trim so the barcodes are maintained for the demultiplexing task